README.adoc

```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
 * to remove & create the local test namespace, and create ClusterRole/ClusterRoleBinding and ServiceAccount inside of the namespace: `$ make reset-namespace`
 
 === End-to-End tests
+
+==== Background & pairing
+
+E2E tests are not located in this repository - all e2e tests are in https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo, however, it's still possible to run them locally from this repo - see <<Running End-to-End Tests>>.
+
+When there is a change introduced in this repository that should be either covered by e2e tests or requires changes in the already existing tests, then all needed changes should go into the https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo.
+The logic that executes tests in openshift-ci automatically tries to pair PR opened for this (member-operator) repository with a branch that potentially exists in the developer's fork of the https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo. This pairing is based on a branch name.
+
+For example, if a developer with GH account `cooljohn` opens a PR (for member-operator repo) from a branch `fix-reconcile`, then the logic checks if there is a branch `fix-reconcile` also in the `cooljohn/toolchain-e2e` fork.
```

Comment:
"For example, if a developer with GH a account `cooljohn`..."

Reply:
"if there is a branch named `fix-reconcile`"

Reply:
not sure if we should use `fork` or `clone` here, but I guess both are fine.

Reply:
@xcoulon i think you meant this: For example, if a developer with _a_ GH ~~a~~ account cooljohn... :-)

Reply:
@xcoulon i think you meant this: For example, if a developer with a GH a account cooljohn... :-)